homeassistant/components/tplink/sensor.py

```diff
@@ -69,43 +142,48 @@ class TPLinkSensorEntityDescription(SensorEntityDescription):
         device_class=SensorDeviceClass.ENERGY,
         state_class=SensorStateClass.TOTAL_INCREASING,
         precision=3,
+        value_fn=async_handle_emeter_attr,
     ),
     TPLinkSensorEntityDescription(
         key=ATTR_VOLTAGE,
         native_unit_of_measurement=UnitOfElectricPotential.VOLT,
         device_class=SensorDeviceClass.VOLTAGE,
         state_class=SensorStateClass.MEASUREMENT,
-        emeter_attr="voltage",
+        attribute_name="voltage",
+        value_fn=async_handle_emeter_attr,
         precision=1,
     ),
     TPLinkSensorEntityDescription(
         key=ATTR_CURRENT_A,
         native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
         device_class=SensorDeviceClass.CURRENT,
         state_class=SensorStateClass.MEASUREMENT,
-        emeter_attr="current",
+        attribute_name="current",
+        value_fn=async_handle_emeter_attr,
         precision=2,
     ),
+    TPLinkSensorEntityDescription(
+        key="on_since",
+        name="On Since",
+        icon="mdi:clock",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        attribute_name="on_since",
+        value_fn=async_handle_timestamp,
+    ),
+    TPLinkSensorEntityDescription(
+        key="wifi_rssi",
+        translation_key="wifi_signal",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        attribute_name="rssi",
+        value_fn=async_handle_int_value,
```

Comment:
We should not add device-level sensors like RSSI for child sockets, add a new field to the description for this case?